Ginger/GingerCoreCommon/Repository/RepositoryFolder.cs

```diff
@@ -696,8 +696,14 @@ public override RepositoryFolderBase AddSubFolder(string folderName)
         /// Save the Repository Item to folder and add it to cache
         /// </summary>
         /// <param name="repositoryItem"></param>
-        public override void AddRepositoryItem(RepositoryItemBase repositoryItem)
+        public override void AddRepositoryItem(RepositoryItemBase repositoryItem, bool addOnlyToCache = false)
         {
+            if(addOnlyToCache)
```

Comment:
instead of this handling, do all lines of codes the original method is doing (because we still want the BF to get the right folder name, added to cacheFolder, etc.) just skip on save line of code if the added "DoNotSave" flag is true, like:
if (!DoNotSave)
{
    SolutionRepository.SaveNewRepositoryItem(repositoryItem);
}